### PR TITLE
Reverse proxy paraview

### LINKED ISSUE #180: Paraview issues
- [x] input-retriever currently hard-coded sets a .vtk extension (this must be changed or other file types will not be readable)
- [ ] needs to use docker secrets
- [x] boot.sh uses the parameter "ws://osparc01.speag.com:..." hard-coded. this should be "ws://localhost:..." in development mode, and something else if running on a different host. (hostname -f is not working)
- [ ] Error when opening paraview viewer (requires #207)
`Failed to load http://osparc01.speag.com:30258/retrieve: No 'Access-Control-Allow-Origin' header is present on the requested resource. Origin 'http://osparc01.speag.com:9081' is therefore not allowed access.`
- [x] If files were removed from S3, they should also be removed from local storage to correctly reflect S3 status

links to #27

### LINKED ISSUE #568: Paraview service not working
Since it complains about the web-socket connection, please, check the changes in the reverse proxy.

### LINKED ISSUE #828: cannot update paraviewweb to latest version it breaks the oSparc theming
it breaks the osparc customisation

### PR DESCRIPTION
**Pull request by sanderegg**
## What do these changes do?

- re-enabled paraview reverse proxy by fixing URL constructions, and use of fixed websocket URL (always on node:port/ws
- director passes an additional SIMCORE_HOSTNAME env variable to each created node containing the node's host name

## Related issue number
fixes #180 
fixes #568 
fixes #828 

## Checklist

- [x] I think the code is well written
- [x] Unit tests for the changes exist
- [x] Documentation reflects the changes
- [x] **Runs in the swarm**
- [x] If you design a new module, add your user to .github/CODEOWNERS

### CODE REVIEW

File: services/web/server/src/simcore_service_webserver/reverse_proxy/handlers/paraview.py [hunk 1]
```diff
@@ -10,20 +10,22 @@
...
+async def handler(req: web.Request, service_url: str, mountpoint: str, proxy_path: str):
     # FIXME:
     assert req.path_qs.endswith(proxy_path)
-    assert mount_point in req.path, "Expected /x/identifier as mount point, got %s" % req.path
+    assert mountpoint in req.path, "Expected /x/identifier as mount point, got %s" % req.path
 
     #target_url = service_url + req.path_qs
-
+                    mountpoint, proxy_path, req.headers)
+    full_url = URL(service_url) / proxy_path
```
Comment (oetiker) [inline services/web/server/src/simcore_service_webserver/reverse_proxy/handlers/paraview.py]: is / a special kind of concatenation?

Comment (sanderegg) [reply services/web/server/src/simcore_service_webserver/reverse_proxy/handlers/paraview.py]: This is  using the yarl library for URL. It is like pathlib but for URLs and brings a bit of flexibility with paths including / or not

Comment (oetiker) [reply services/web/server/src/simcore_service_webserver/reverse_proxy/handlers/paraview.py]: operator overloading ... cool!

File: services/web/client/source/class/qxapp/data/model/Node.js [hunk 2]
```diff
@@ -760,8 +760,6 @@ qx.Class.define("qxapp.data.model.Node", {
         // FIXME: this is temporary until the reverse proxy works for these services
         if (this.getKey().includes("neuroman") || this.getKey().includes("modeler")) {
           srvUrl = "http://" + window.location.hostname + ":" + publishedPort + srvUrl;
-        } else if (this.getKey().includes("3d-viewer")) {
-          srvUrl = "http://" + window.location.hostname + ":" + publishedPort + entryPoint;
         }
```
Comment (odeimaiz) [inline services/web/client/source/class/qxapp/data/model/Node.js]: one hack less 👍

Comment (sanderegg) [reply services/web/client/source/class/qxapp/data/model/Node.js]: will then only work from version 2.11.0 of paraview, I did not try other versions.

**Pull request merged**